### PR TITLE
Oak-10137 | Fix static variable waitForESAcknowledgement being accessed from non static context

### PR DESCRIPTION
**Pull request by nit0906**

### CODE REVIEW

File: oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticBulkProcessorHandler.java [hunk 1]
```diff
@@ -64,13 +64,13 @@ class ElasticBulkProcessorHandler {
...
     private static final String SYNC_MODE_PROPERTY = "sync-mode";
     private static final String SYNC_RT_MODE = "rt";
-    private static boolean waitForESAcknowledgement = true;
 
     protected final ElasticConnection elasticConnection;
     protected final String indexName;
     protected final ElasticIndexDefinition indexDefinition;
     private final NodeBuilder definitionBuilder;
     protected final BulkProcessor bulkProcessor;
+    private boolean waitForESAcknowledgement;
```
Comment (fabriziofortino) [inline oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticBulkProcessorHandler.java]: make it final

File: oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticBulkProcessorHandler.java [hunk 2]
```diff
@@ -96,11 +96,13 @@ class ElasticBulkProcessorHandler {
     private ElasticBulkProcessorHandler(@NotNull ElasticConnection elasticConnection,
                                         @NotNull String indexName,
                                         @NotNull ElasticIndexDefinition indexDefinition,
-                                        @NotNull NodeBuilder definitionBuilder) {
+                                        @NotNull NodeBuilder definitionBuilder,
+                                        @NotNull boolean waitForESAcknowledgement) {
```
Comment (fabriziofortino) [inline oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticBulkProcessorHandler.java]: primitive should not be annotated with `@NotNull`: remove the annotationor use Boolean

File: oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticBulkProcessorHandler.java [hunk 3]
```diff
@@ -113,20 +115,12 @@ private ElasticBulkProcessorHandler(@NotNull ElasticConnection elasticConnection
     public static ElasticBulkProcessorHandler getBulkProcessorHandler(@NotNull ElasticConnection elasticConnection,
                                                                       @NotNull String indexName,
                                                                       @NotNull ElasticIndexDefinition indexDefinition,
-                                                                      @NotNull NodeBuilder definitionBuilder, CommitInfo commitInfo) {
+                                                                      @NotNull NodeBuilder definitionBuilder, CommitInfo commitInfo,
+                                                                      @NotNull boolean waitForESAcknowledgement) {
```
Comment (fabriziofortino) [inline oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticBulkProcessorHandler.java]: see above comment on annotation for primitives

File: oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticBulkProcessorHandler.java [hunk 4]
```diff
@@ -300,8 +294,9 @@ protected static class RealTimeBulkProcessorHandler extends ElasticBulkProcessor
         private RealTimeBulkProcessorHandler(@NotNull ElasticConnection elasticConnection,
                                              @NotNull String indexName,
                                              @NotNull ElasticIndexDefinition indexDefinition,
-                                             @NotNull NodeBuilder definitionBuilder) {
-            super(elasticConnection, indexName, indexDefinition, definitionBuilder);
+                                             @NotNull NodeBuilder definitionBuilder,
+                                             @NotNull boolean waitForESAcknowledgement) {
```
Comment (fabriziofortino) [inline oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticBulkProcessorHandler.java]: see above comment on annotation for primitives

**Review by fabriziofortino — APPROVED** [review]
refactoring looks okay. Just a few minors

**Review by nfsantos — APPROVED** [review]
LGTM

**Pull request merged**